internal/tui/components/issueview/issueview.go

```diff
@@ -97,22 +114,64 @@ func (m Model) Update(msg tea.Msg) (Model, tea.Cmd) {
 		} else if m.isLabeling {
 			switch msg.Type {
 			case tea.KeyCtrlD:
-				labels := strings.Fields(m.inputBox.Value())
+				labels := m.inputBox.GetAllLabels()
 				if len(labels) > 0 {
 					cmd = m.label(labels)
 				}
 				m.inputBox.Blur()
 				m.isLabeling = false
+				m.ac.Hide()
 				return m, cmd
 
 			case tea.KeyEsc, tea.KeyCtrlC:
 				m.inputBox.Blur()
 				m.isLabeling = false
+				m.ac.Hide()
+				return m, nil
+
+			case tea.KeyUp, tea.KeyCtrlP:
+				m.ac.Prev()
+				return m, nil
+
+			case tea.KeyDown, tea.KeyCtrlN:
+				m.ac.Next()
+				return m, nil
+
+			case tea.KeyTab, tea.KeyEnter:
+				selected := m.ac.GetSelected()
+				if selected != "" {
+					currentInput := m.inputBox.Value()
+					currentLabel := m.inputBox.GetCurrentLabel()
+					if currentLabel != "" {
+						lastComma := strings.LastIndex(currentInput, ",")
+						if lastComma == -1 {
+							// No comma, replace entire string
+							m.inputBox.SetValue(selected + ", ")
+						} else {
+							// Replace only the text after the last comma
+							previousLabels := currentInput[:lastComma+1]
+							m.inputBox.SetValue(previousLabels + " " + selected + ", ")
+						}
+					} else {
+						// No current label, append suggestion
+						newValue := m.inputBox.Value()
+						if newValue != "" && !strings.HasSuffix(newValue, ", ") {
+							newValue += ", "
+						}
+						m.inputBox.SetValue(newValue + selected + ", ")
+					}
+				}
+				m.ac.Hide()
 				return m, nil
 			}
 
 			m.inputBox, taCmd = m.inputBox.Update(msg)
 			cmds = append(cmds, cmd, taCmd)
+
+			currentInput := m.inputBox.Value()
+			existingLabels := m.inputBox.GetAllLabels()
+			m.ac.Filter(currentInput, existingLabels)
```

Comment:
I don't think it's clear that `.Filter` will show the autocomplete.
I would expect `.Show` to do that. A bit confusing as I see various calls to `.Hide` but none to `.Show`.